add support for type_as/_to_copy

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #78524
* #78523
* #78522
* #78517
* #78270
* #78090
* #78516
* #77972
* __->__ #77971
* #77970
* #77969

Without this I would get strange `invalid arguments` in CUDA call.

Commit:
add support for type_as/_to_copy

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

ezyang commented on torch/_subclasses/fake_tensor.py:
```diff
@@ -60,6 +61,17 @@ def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
         # Run the original computation
 
+        # _to_copy fails when run with FakeTensors to cuda device
+        # TODO: debug
```
You're talking about the meta implementation here right? It makes sense, since we can't actually produce the cuda data (but for fake tensor that's OK, because the fake cuda device doesn't have data anyway)

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36618463).

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

Differential Revision: [D36618463](https://our.internmc.facebook.com/intern/diff/D36618463)

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36618463).

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

Differential Revision: [D36618463](https://our.internmc.facebook.com/intern/diff/D36618463)

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

Differential Revision: [D36618463](https://our.internmc.facebook.com/intern/diff/D36618463)

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

Differential Revision: [D36618463](https://our.internmc.facebook.com/intern/diff/D36618463)

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.

Differential Revision: [D36618463](https://our.internmc.facebook.com/intern/diff/D36618463)

[ghstack-poisoned]

Commit:
Update on "add support for type_as/_to_copy"


Without this I would get strange `invalid arguments` in CUDA call.



[ghstack-poisoned]

eellison commented:
@pytorchbot merge this please
